### PR TITLE
[2.x] Adding zip support for directories

### PR DESCRIPTION
**Pull request by kasvith**
## Purpose
To enable apimcli to archive a directory in a given path and import it to APIM

## Approach
- Check given path exists and it is a directory
- If directory apimcli will pack an archive with directory contents
- Use new archive to import API

## User stories
- Users now can edit and easily import APIs without manually zipping it

## Test environment
- Linux 4.15.0-46-generic #49-Ubuntu SMP  x86_64 GNU/Linux

### CODE REVIEW

File: import-export-cli/cmd/importAPI.go [hunk 1]
```diff
@@ -114,10 +114,35 @@ func ImportAPI(query, apiImportExportEndpoint, accessToken, exportDirectory stri
...
+	fileName := query
+	zipFilePath := fileName
 
-	var zipFilePath string = fileName
-	// Test if we can find the file in the current work directory
+	// Check whether the given path is a directory
+	// If it is a directory archive it
+	if info, err := os.Stat(fileName); err == nil && info.IsDir() {
+		fmt.Println(fileName + " is a directory")
+		fmt.Println("Creating an archive from the directory...")
```
Comment (ruks) [inline import-export-cli/cmd/importAPI.go]: Seems there are many blank lines. Can we remove them?

Comment (kasvith) [reply import-export-cli/cmd/importAPI.go]: I made some improvements to comments and others, also removed some blank lines.
But I think these lines help understanding code faster :)

Comment (ruks) [reply import-export-cli/cmd/importAPI.go]: please remove the blank lines in other places. including zipUtils.go file. Keeping the blank line is ok sometimes, But better to keep comments and remove blank line. so we can reduce unnecessary line count of the file.

Comment (kasvith) [reply import-export-cli/cmd/importAPI.go]: sure

File: import-export-cli/utils/zipUtils.go [hunk 2]
```diff
@@ -19,23 +19,69 @@
...
+		return nil
+	}
+
+	var baseDir string
+	if info.IsDir() {
+		baseDir = filepath.Base(source)
+	}
+
+	filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```
Comment (harsha89) [inline import-export-cli/utils/zipUtils.go]: better to use meanigful vairable names

File: import-export-cli/utils/zipUtils.go [hunk 3]
```diff
@@ -19,23 +19,69 @@
...
+			return err
+		}
+
+		header, err := zip.FileInfoHeader(info)
```
Comment (harsha89) [inline import-export-cli/utils/zipUtils.go]: add some comments and debug logs if possible

File: import-export-cli/utils/zipUtils.go [hunk 4]
```diff
@@ -19,23 +19,69 @@
...
+	if err != nil {
+		return nil
+	}
+
+	var baseDir string
+	if info.IsDir() {
+		baseDir = filepath.Base(source)
+	}
+
+	filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
```
Comment (harsha89) [inline import-export-cli/utils/zipUtils.go]: what does this fucntion do and add it as a comment

**Pull request merged**